### PR TITLE
Art 6076 inform user when an update is available

### PR DESCRIPTION
**Pull request by korijHT**
When performing a command action the CLI will check for updates.
If a newer version exists then the user will be prompted to update.
If they choose not to, they will then be prompted with an option to disable this feature.
There is now a "check-for-updates" Global Setting that persists and can be toggled.

### CODE REVIEW

File: UpdateService/UpdateService.cs [hunk 1]
```diff
@@ -18,49 +17,43 @@ namespace Hypertherm.Update
...
+            GetLatestReleasedVersionInfo().GetAwaiter().GetResult();
         }
 
-        public async Task<bool> IsUpdateAvailable()
+        public bool IsUpdateAvailable()
         {
-            _analyticsService.GenericTrace($"Checking for updates.");
-
-            var latestVersion = new Version("0.0.0.0");
+            _analyticsService.GenericTrace($"Checking for available updates.");
```
Comment (Jjohns-Hypertherm) [inline UpdateService/UpdateService.cs:51]:
```suggestion
            _analyticsService.GenericTrace($"Checking for updates.");

```

File: UpdateService/UpdateService.cs [hunk 2]
```diff
@@ -147,11 +140,29 @@ public async Task Update(string version = "latest")
...
                 }
-                
+
                 ExecuteCommand(tmpDir + update, currentDir);
             }
         }
 
+        private async Task GetLatestReleasedVersionInfo()
+        {
+            _analyticsService.GenericTrace($"Acquiring the latest release's version info.");
```
Comment (Jjohns-Hypertherm) [inline UpdateService/UpdateService.cs:150]:
```suggestion
            _analyticsService.GenericTrace($"Getting the latest release's version information.");

```

File: cc-cli/ArgumentHandler.cs [hunk 3]
```diff
@@ -219,7 +235,9 @@ public bool IsFilenameAKeyword(string filename)
             ["-u"] = "Units",
             ["--units"] = "Units",
             ["-t"] = "CcType",
-            ["--type"] = "CcType"
+            ["--type"] = "CcType",
```
Comment (Jjohns-Hypertherm) [inline cc-cli/ArgumentHandler.cs:238]: Is "CcType" customer facing? Wondering because the ReadMe just says "type." If we are changing it to "CcType", what is the reason for the change?

Comment (korijHT) [reply cc-cli/ArgumentHandler.cs:238]: This didn't change, what you are seeing is the addition of a comma.
We added a new option "-s", so a comma is needed to seperate them.
"CcType" is not user facing it is related to a property name.

File: UpdateService/UpdateService.cs (hunk 1 above)
Comment (korijHT) [reply UpdateService/UpdateService.cs:50]: Yea this looks good. I might even go as far as changing updates to singular update.
Keep in mind _analyticsService is a logging interface, for us it is using Application Insights.
